### PR TITLE
[TS-352]<fix>taosdemo support unsigned data type as well as negative data

### PR DESCRIPTION
**Pull request by zhaoyanggh**

### CODE REVIEW

File: src/kit/taosdemo/taosdemo.c [hunk 1]
```diff
@@ -6427,8 +6427,7 @@ static int64_t generateStbRowData(
             "(%" PRId64 "", timestamp);
 
     for (int i = 0; i < stbInfo->columnCount; i++) {
-        tstrncpy(pstr + dataLen, ",", 2);
-        dataLen += 1;
+        dataLen += snprintf(pstr + dataLen, 2, ",");
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c]: printf family function has longer execution time lead performance decrease.

File: src/kit/taosdemo/taosdemo.c [hunk 2]
```diff
@@ -6462,22 +6461,25 @@ static int64_t generateStbRowData(
                         tmp = rand_int_str();
                     }
                     tmpLen = strlen(tmp);
-                    tstrncpy(pstr + dataLen, tmp, min(tmpLen + 1, INT_BUFF_LEN));
+                    dataLen += snprintf(pstr + dataLen, tmpLen + 1, "%s", tmp);
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c]: printf family function has longer execution time lead performance decrease.

**Review by sangshuduo — CHANGES_REQUESTED** [review]

File: src/kit/taosdemo/taosdemo.c (hunk 1 above)
Comment (zhaoyanggh) [reply src/kit/taosdemo/taosdemo.c]: I will change it back

File: src/kit/taosdemo/taosdemo.c [hunk 3]
```diff
@@ -2382,9 +2483,19 @@ static void init_rand_data() {
...
+    assert(g_randutinyint_buff);
+    g_randusmallint_buff = calloc(1, SMALLINT_BUFF_LEN * MAX_PREPARED_RAND);
+    assert(g_randusmallint_buff);
+    g_randubigint_buff = calloc(1, BIGINT_BUFF_LEN * MAX_PREPARED_RAND);
+    assert(g_randubigint_buff);
 
     for (int i = 0; i < MAX_PREPARED_RAND; i++) {
-        g_randint[i] = (int)(taosRandom() % 65535);
+        // TODO: make g_randint list include negative number
+        g_randint[i] = (int)(taosRandom() % 131069 - 65534);
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c]: how about use "taosRandom() % RAND_MAX - (RAND_MAX >> 1) “  to extend max random value to all possible integer value?

Comment (sangshuduo) [reply src/kit/taosdemo/taosdemo.c]: we can remove TODO after negative value generated

Comment (zhaoyanggh) [reply src/kit/taosdemo/taosdemo.c]: sounds good, will make our data more real.

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**